waku/discovery/waku_discv5.nim

```diff
@@ -373,3 +373,28 @@ proc setupDiscoveryV5*(
   WakuDiscoveryV5.new(
     rng, discv5Conf, some(myENR), some(nodePeerManager), nodeTopicSubscriptionQueue
   )
+
+proc updateBootstrapRecords*(
```

Comment:
is there any value in updating bootstrap nodes on runtime? aren't they only used when connecting to the network?

Reply:
> is there any value in updating bootstrap nodes on runtime? aren't they only used when connecting to the network?

Good point! This feature comes from what we are already offering from `go-waku` so we offer the same. I guess that was a requirement from any of the current projects that integrates us (cc @richard-ramos .)

Reply:
> dns discovery

are there nodes that do not require dns discovery in the bootstrap node list?

Reply:
> > dns discovery
> 
> are there nodes that do not require dns discovery in the bootstrap node list?

In case you asked w.r.t `wakunode2` (`nwaku`), it is mandatory for the app to retrieve bootstrap nodes from DNS:
https://github.com/waku-org/nwaku/blob/5ee4cba5340ac64223e72c7abb204da6944af9de/waku/factory/waku.nim#L140-L161

Reply:
don't know the details then, but that sounds like an unnecessary single point of failure - normally, one would bootstrap discv5 from multiple sources (dns, hardcoded nodes, nodes from previous startups etc)

Reply:
My understanding is that it's not mandatory. If we don't have DNS discovery enabled `retrieveDynamicBootstrapNodes` will return `ok` with an empty seq and the node will simply set up without dynamic bootstrap nodes.

https://github.com/waku-org/nwaku/blob/5ee4cba5340ac64223e72c7abb204da6944af9de/waku/discovery/waku_dnsdisc.nim#L130-L131

However, if we enable DNS discovery and it fails retrieving the nodes then the whole node setup fails. Not sure if it should be like that or if it would be better to add a warning and continue the setup using other sources as a backup (as if DNS discovery wasn't configured)